src/physics/cam/clubb_intr.F90

```diff
@@ -1003,6 +1008,9 @@ subroutine clubb_ini_cam(pbuf2d)
     qist_idx    = pbuf_get_index('QIST')        ! Physical in-stratus IWC
     dp_frac_idx = pbuf_get_index('DP_FRAC')     ! Deep convection cloud fraction
     icwmrdp_idx = pbuf_get_index('ICWMRDP')     ! In-cloud deep convective mixing ratio
+!+++MKW
+    icwmrsh_idx = pbuf_get_index('ICWMRSH')     ! In-cloud shallow convective mixing ratio (EDMF)
+!---MKW

```

Comment:
do you know where this pbuf variable is being declared/initialized? convect_shallow? Can we confirm it's being initialized somewhere to zero?

Reply:
Most of convect_shallow is irrelevant when CLUBB is on, but I tried to do an addfld('ICWMRSH'...) in clubb_mf and the compiler got mad at me so convect_shallow is the obvious place. I'll look into it and report back here.

Reply:
I took a quick look, and ICWMRSH is being declared in convect_shallow_register, and set to zero in convect_shallow_tend. This is fine because clubb_intr is downstream of convect_shallow, so it will overwrite the zero being set there. This is exactly how I handle prec_sh, since it is also set in convect_shallow to zero, but since clubb_intr is downstream, it just overwrites it to the MF precip. *HOWEVER* in convect_shallow it is also doing an outfld for icwmrsh:
https://github.com/ESCOMP/CAM/blob/b1d22e5cd63b1e66911598a666fb630028146f22/src/physics/cam/convect_shallow.F90#L740

So this needs to be commented out for clarity. With prec_sh I got lucky b/c it only outflds it for non-clubb shallow schemes, but this conditional is not true for icwmrsh.

Reply:
OK I commented the outfld for ICWMRSH in convect_shallow and added one in clubb_intr.

Reply:
great. Ill resolve these convos once you commit/push changes back to your branch.

Reply:
Push complete